programming_exercises/personalized_exercises.rst

```diff
@@ -15,15 +15,14 @@ and multiple students may be assigned the same instance. The instance assigned
 to the student remains the same even after refreshing the exercise page.
 
 A personalized exercise defines a generator program that generates N instances
-to the exercise. The instances are generated before the start of the course.
-The mooc-grader CLI command ``python manage.py pregenerate_exercises``
-runs the generator programs and saves the instance files in the correct
-directory under the mooc-grader. In theory, the instance files could also
-be created by some other way in the correct location.
+to the exercise. The instances should be generated before the start of the course.
+The Python script `pregenerate_exercises.py <https://github.com/apluslms/aplus-manual/blob/master/pregenerate_exercises.py>`_
+can be used by teachers and assistants to
+run the generator programs and to save the instance files in the directory
+``personalized_exercises`` under the course Git repository root. In theory,
+the instance files could also be created by some other way in the correct location.
 The CLI command has some arguments to modify its behaviour: they are printed
-with the ``--help`` argument. The ``run-mooc-grader`` container (used in local
-testing and development) should automatically call the generators
-if there are any personalized exercises in the course.
+with the ``--help`` argument.
```

Comment:
Because this script is now intended to be run by regular teacher-users with varying levels of Python experience (in contrast to sysadmins as earlier), I would add more concrete use examples of the script:

`python3 ./pregenerate_exercises.py` -- generate all personalized exercises

`python3 ./pregenerate_exercises.py exercise_path` -- generate only those personalize exercises that are in exercise_path

`python3 ./pregenerate_exercises.py --help` -- show all command line attributes

Also here could be a good place to mention that pyyaml library may need to be installed, if it does not yet exist:
`pip3 install pyyaml`

Also, I suppose the idea is to push the generated exercises to git repo after generation, so that the gitmanager/grader can find them at the server end. That should be mentioned also, again with some examples:

    git add personalized_exercises/*
    git commit -m"Personalized exercises generated"
    git push